### PR TITLE
Doc 12037 vector search beta2 dotnet

### PR DESCRIPTION
**Pull request by ElliotFrancisHunter**
This PR contains the doc copy for the upcoming Vector Search .NET beta 2 release.

It also has some small fly by fixes:

- Removal of British English on existing Vector Search pages
- Reorder of release notes and upgrade links on what's new to reflect order on left hand nav
- A handful of fixed typos

### CODE REVIEW

File: modules/csharp/pages/gs-build.adoc [hunk 1]
```diff
@@ -30,6 +30,31 @@ The document was successfully persisted to the database.
...
+using Couchbase.Lite.Extensions;
+
+----
++
+. Load the extension.
++
+[source, csharp]
+----
+// For Android applications
+Couchbase.Lite.Support.Droid.Activate(Context);
```
Comment (pasin) [inline modules/csharp/pages/gs-build.adoc]: This line is not required for loading the vector search extension but it's required for using .NET on the android platform in general. With the assumption that this is already explained somewhere else, this line can be removed here.

File: modules/csharp/pages/gs-install.adoc [hunk 2]
```diff
@@ -58,10 +68,38 @@ This is the recommended method of dependency management because it supports the
...
+* `Couchbase.Lite.Enterprise.Support.android`
+* `Couchbase.Lite.Enterprise.Support.NetDesktop`
+* `Couchbase.Lite.Enterprise.Support.WinUI`
+
+.`Couchbase.Lite.VectorSearch`
+* `Couchbase.Lite.VectorSearch.Support.UWP`
+* `Couchbase.Lite.VectorSearch.Support.ios`
+* `Couchbase.Lite.VectorSearch.Support.android`
+* `Couchbase.Lite.VectorSearch.Support.NetDesktop`
+* `Couchbase.Lite.VectorSearch.Support.WinUI`
```
Comment (pasin) [inline modules/csharp/pages/gs-install.adoc]: I don't think this list is correct. @borrrden Can you confirm?

**Review by pasin — COMMENTED** [review]
Need @borrrden to confirm about the supported platforms in the vector search package. The section describing the key concepts is very useful and easy to understand.

File: modules/csharp/pages/gs-install.adoc (hunk 2 above)
Comment (pasin) [reply modules/csharp/pages/gs-install.adoc]: As @borrrden reminded me on slack about this, I think the package will just be "Couchbase.Lite.VectorSearch" and there are no Support packages needed to be installed.

File: modules/csharp/examples/code_snippets/VectorSearch.cs [hunk 3]
```diff
@@ -67,14 +66,14 @@ private void CreateCustomVectorIndexConfig()
...
-            // end::vs - create - default - config[]
+            // end::vs-create-custom-config[]
         }
 
         private void CreateVectorIndex()
         {
-            var database = new Database("my-database");
-
             // tag::vs-create-index[]
+            var database = new Database("my-database");
```
Comment (borrrden) [inline modules/csharp/examples/code_snippets/VectorSearch.cs:75]: In general I want to leave lines like this out of the snippets. It should be obvious that you need to create a database, and this just adds needless lines to the snippet.

File: modules/csharp/examples/code_snippets/VectorSearch.cs [hunk 4]
```diff
@@ -171,11 +170,12 @@ private void QueryUsingVectorMatch()
 
         private void QueryUsingVectorDistance()
         {
+            // tag::vs-use-vector-distance[]
```
Comment (borrrden) [inline modules/csharp/examples/code_snippets/VectorSearch.cs]: Ditto here, move the `var database` line out of the tag.  Also there is no real need to make the `inputColor` variable here, it could simply be used as a literal later.

File: modules/csharp/pages/gs-build.adoc [hunk 5]
```diff
@@ -30,6 +30,31 @@ The document was successfully persisted to the database.
...
+Extension.Load(new VectorSearchExtension(Context));
```
Comment (borrrden) [inline modules/csharp/pages/gs-build.adoc:51]: Warning:  These will not get checked by the compiler in the way that snippets will.

File: modules/csharp/pages/gs-install.adoc [hunk 6]
```diff
@@ -58,10 +68,38 @@ This is the recommended method of dependency management because it supports the
...
+Vector Search is available only for 64-bit architectures and
+Intel processors that support the Advanced Vector Extensions 2 (AVX2) instruction set.
+To verify whether your device supports the AVX2 instructions set, https://www.intel.com/content/www/us/en/support/articles/000090473/processors/intel-core-processors.html[follow these instructions.] 
+--
+
+==== Package Reference
+This is the recommended method of dependency management because it supports the strict version requirement between the core _Couchbase Lite for .Net_ package and its dependent Support library, which comprises:
+
+.`Couchbase.Lite.Enterprise`
+* `Couchbase.Lite.Enterprise.Support.UWP`
```
Comment (borrrden) [inline modules/csharp/pages/gs-install.adoc]: Remove this, this platform is removed as of 3.2

File: modules/csharp/pages/gs-install.adoc [hunk 7]
```diff
@@ -58,10 +68,38 @@ This is the recommended method of dependency management because it supports the
...
+To use Vector Search, you must have Couchbase Lite installed and add the Vector Search extension to your Couchbase Lite application.
+Vector Search is available only for 64-bit architectures and
+Intel processors that support the Advanced Vector Extensions 2 (AVX2) instruction set.
+To verify whether your device supports the AVX2 instructions set, https://www.intel.com/content/www/us/en/support/articles/000090473/processors/intel-core-processors.html[follow these instructions.] 
+--
+
+==== Package Reference
+This is the recommended method of dependency management because it supports the strict version requirement between the core _Couchbase Lite for .Net_ package and its dependent Support library, which comprises:
+
+.`Couchbase.Lite.Enterprise`
```
Comment (borrrden) [inline modules/csharp/pages/gs-install.adoc]: Does this really need to be duplicated here?  Aren't these install instructions already somewhere else for the main package?  This is making this page overly verbose imo.

**Review by borrrden — DISMISSED** [review]

File: modules/csharp/pages/gs-install.adoc (hunk 6 above)
Comment (ElliotFrancisHunter) [reply modules/csharp/pages/gs-install.adoc]: Done.

File: modules/csharp/pages/gs-install.adoc (hunk 2 above)
Comment (ElliotFrancisHunter) [reply modules/csharp/pages/gs-install.adoc]: Okay I've removed the support packages in this section for Couchbase.Lite.VectorSearch. I'm assuming you still need the Enterprise packages for CBL so have left them as is.

File: modules/csharp/pages/gs-install.adoc (hunk 7 above)
Comment (ElliotFrancisHunter) [reply modules/csharp/pages/gs-install.adoc]: I see your point. I've condensed to a couple of lines but will keep the section to make sure Vector Search instructions are distinguishable.

File: modules/csharp/pages/gs-build.adoc (hunk 1 above)
Comment (ElliotFrancisHunter) [reply modules/csharp/pages/gs-build.adoc]: Done!

File: modules/csharp/pages/gs-build.adoc (hunk 5 above)
Comment (ElliotFrancisHunter) [reply modules/csharp/pages/gs-build.adoc:51]: Done!

File: modules/csharp/examples/code_snippets/VectorSearch.cs (hunk 4 above)
Comment (ElliotFrancisHunter) [reply modules/csharp/examples/code_snippets/VectorSearch.cs]: Done!

File: modules/ROOT/pages/cbl-whatsnew.adoc [hunk 8]
```diff
@@ -17,6 +17,27 @@ NOTE: Couchbase Lite 3.0 introduces some breaking changes. +
...
+
+=== New Features
+
+==== Vector Search
+
+[IMPORTANT] 
+--
+Vector Search is available only for 64-bit architectures and
+Intel processors that support the Advanced Vector Extensions 2 (AVX2) instruction set.
+To verify whether your device supports the AVX2 instructions set, https://www.intel.com/content/www/us/en/support/articles/000090473/processors/intel-core-processors.html[follow these instructions.]
```
Comment (lamagnifica) [inline modules/ROOT/pages/cbl-whatsnew.adoc:30]: instruction singular here?

File: modules/csharp/pages/gs-build.adoc [hunk 9]
```diff
@@ -30,6 +30,30 @@ The document was successfully persisted to the database.
 
 See also -- {sgw-install--xref}
 
+=== Building With Vector Search
+
+To build your Couchbase Lite application with Vector Search, you must add the following to your existing Couchbase Lite application:
```
Comment (lamagnifica) [inline modules/csharp/pages/gs-build.adoc:35]: "add" doesn't seem like the right verb here, since you're adding, loading...
"complete" the following for your...?

File: modules/csharp/pages/gs-install.adoc [hunk 10]
```diff
@@ -29,12 +29,22 @@ For experienced developers, this is all you need to add _Couchbase Lite for {par
...
+Vector Search is available only for 64-bit architectures and
+Intel processors that support the Advanced Vector Extensions 2 (AVX2) instruction set.
+To verify whether your device supports the AVX2 instructions set, https://www.intel.com/content/www/us/en/support/articles/000090473/processors/intel-core-processors.html[follow these instructions.] 
+--
 ****
 
 That's it! +
 You should be ready to build you app using this version.
-The rest of this content contains more detail, for those who want tp know more about the install or who encountered issues
+The rest of this content contains more detail, for those who want to know more about the install or who encountered issues
```
Comment (lamagnifica) [inline modules/csharp/pages/gs-install.adoc:47]: add period at end?

File: modules/csharp/pages/gs-install.adoc [hunk 11]
```diff
@@ -45,31 +55,59 @@ Couchbase recommends installing Nuget packages via _PackageReference_.
...
 .`Couchbase.Lite`
-* `Couchbase.Lite.Support.UWP`
 * `Couchbase.Lite.Support.ios`
 * `Couchbase.Lite.Support.android`
 * `Couchbase.Lite.Support.NetDesktop`
 * `Couchbase.Lite.Support.WinUI`
 
+=== Installing Vector Search 
+
+Couchbase recommends installing Nuget packages via _PackageReference_.
```
Comment (lamagnifica) [inline modules/csharp/pages/gs-install.adoc:71]: NuGet (?)

Comment (lamagnifica) [reply modules/csharp/pages/gs-install.adoc:71]: also, italics are bad according to the style guide

File: modules/csharp/pages/gs-install.adoc [hunk 12]
```diff
@@ -45,31 +55,59 @@ Couchbase recommends installing Nuget packages via _PackageReference_.
...
+
+[IMPORTANT]
+--
+To use Vector Search, you must have Couchbase Lite installed and add the Vector Search extension to your Couchbase Lite application.
+Vector Search is available only for 64-bit architectures and
+Intel processors that support the Advanced Vector Extensions 2 (AVX2) instruction set.
+To verify whether your device supports the AVX2 instructions set, https://www.intel.com/content/www/us/en/support/articles/000090473/processors/intel-core-processors.html[follow these instructions.]
```
Comment (lamagnifica) [inline modules/csharp/pages/gs-install.adoc:78]: same singular issue

File: modules/csharp/pages/gs-install.adoc [hunk 13]
```diff
@@ -45,31 +55,59 @@ Couchbase recommends installing Nuget packages via _PackageReference_.
...
+[IMPORTANT]
+--
+To use Vector Search, you must have Couchbase Lite installed and add the Vector Search extension to your Couchbase Lite application.
+Vector Search is available only for 64-bit architectures and
+Intel processors that support the Advanced Vector Extensions 2 (AVX2) instruction set.
+To verify whether your device supports the AVX2 instructions set, https://www.intel.com/content/www/us/en/support/articles/000090473/processors/intel-core-processors.html[follow these instructions.] 
+--
+
+==== Vector Search Package Reference
+This is the recommended method of dependency management for the Vector Search extension because it supports the strict version requirement between the core _Couchbase Lite for .Net_ package and its dependent Support library.
```
Comment (lamagnifica) [inline modules/csharp/pages/gs-install.adoc:82]: this is a little hard to follow. Maybe,

Due to the strict version requirement between the core Couchbase Lite for .Net package and its dependent Support library, Couchbase recommends the following dependency management procedure for the Vector Management Extension.
To manage your dependencies:

File: modules/csharp/pages/vector-search.adoc [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+Vector search is a sophisticated data retrieval technique that focuses on matching the contextual meanings of search queries and data entries, rather than simple text matching.
+Vectors are represented by arrays of numbers known as an embedding, which are generated by Large Language Models (LLMs) to depict objects such as text, images, and audio. 
+
+Once you choose the LLM you wish to integrate in your application, you can create vector indexes that will store these embeddings for improved search performance and start querying against them.  
+
+== Applications of Vector Search
+
+You can xref:csharp:gs-install.adoc[use Vector Search] to enhance your mobile and edge applications in a variety of use cases, these include:
+
+* Perform Semantic and Similarity Search on the Edge - Any offline-first mobile or IoT application can benefit from rich semantic text capabilites offered by Vector Search to retrieve contextually relevant data and present it to users.
```
Comment (dgnicholson) [inline modules/csharp/pages/vector-search.adoc]: Nit:
```suggestion
* Perform Semantic and Similarity Search on the Edge - Any offline-first mobile or IoT application can benefit from rich semantic text capabilities offered by Vector Search to retrieve contextually relevant data and present it to users.
```

File: modules/csharp/pages/vector-search.adoc [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+:page-role:
+:description: Use Vector Search to build adaptive and user-focused applications using Generative AI.
+:keywords: edge AI api swift ios macos apple vector search generative
+
+[abstract]
+{description}
+
+== About Vector Search
+
+Vector search is a technique to retrieve semantically similar items based on vector embedding representations of the items in a multi-dimensional space.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:16]: is this Vector search in general, vs. Vector Search which is the Couchbase extension to Lite? or should this be icapped?

**Review by osfameron — APPROVED** [review]
Looks ok to me (I didn't do a very in depth dive, can look closer tomorrow if required)

File: modules/csharp/pages/vector-search.adoc (hunk 15 above)
Comment (lamagnifica) [reply modules/csharp/pages/vector-search.adoc:16]: (the changes in capitalization for Vector S/search are distracting in the next few lines)

File: modules/csharp/pages/vector-search.adoc [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+You can use Vector Search to find the top N items similar to a given item based on their vector representations. 
+Vector Search is an essential component of Generative AI and Predictive AI applications.
+
+Vector search is a sophisticated data retrieval technique that focuses on matching the contextual meanings of search queries and data entries, rather than simple text matching.
+Vectors are represented by arrays of numbers known as an embedding, which are generated by Large Language Models (LLMs) to depict objects such as text, images, and audio.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:21]: I think embeddings plural here?

File: modules/csharp/pages/vector-search.adoc [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+Additionally, Vector Search in Couchbase Lite provides the following benefits:
+
+* Unified Cloud-to-Edge Support for Vector Similarity Search - Couchbase supports Vector Search from cloud to edge, which enables applications to efficiently utilize cloud and edge computing's strengths.
+
+* Enhanced Data Privacy on the Edge - By performing Vector Search within the device, personal data and search queries of a sensitive nature do not have to leave the device.
+
+* Low Latency Application Support - You can run searches locally against a local dataset using a local embedded model. This eliminates the network variability and results in more consistent execution speed.
+Even in the case where the model is not embedded within the local device but is deployed at the edge location, the round trip time (RTT) associated with queries can be significantly reduced compared to searches made over the Internet.
+
+* Cost Per Query Reduction - When you have hundreds of thousands of connected clients querying against a cloud-based LLM, the load on cloud model and operational costs of running the cloud based model can be considerably high.
```
Comment (dgnicholson) [inline modules/csharp/pages/vector-search.adoc:47]:
```suggestion
* Cost Per Query Reduction - When you have hundreds of thousands of connected clients querying against a cloud-based LLM, the load and operational costs of running the cloud based model can be considerably high.
```
Nit: Maybe some clarification

File: modules/csharp/pages/vector-search.adoc [hunk 18]
```diff
@@ -0,0 +1,137 @@
...
+Vector Search is an essential component of Generative AI and Predictive AI applications.
+
+Vector search is a sophisticated data retrieval technique that focuses on matching the contextual meanings of search queries and data entries, rather than simple text matching.
+Vectors are represented by arrays of numbers known as an embedding, which are generated by Large Language Models (LLMs) to depict objects such as text, images, and audio. 
+
+Once you choose the LLM you wish to integrate in your application, you can create vector indexes that will store these embeddings for improved search performance and start querying against them.  
+
+== Applications of Vector Search
+
+You can xref:csharp:gs-install.adoc[use Vector Search] to enhance your mobile and edge applications in a variety of use cases, these include:
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:27]: two sentences:
"...cases. These include:"

File: modules/csharp/pages/vector-search.adoc [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+
+* Perform Semantic and Similarity Search on the Edge - Any offline-first mobile or IoT application can benefit from rich semantic text capabilites offered by Vector Search to retrieve contextually relevant data and present it to users.
+
+* Create Recommendation Engines - Vector Search enables the creation of advanced recommendation systems that analyze semantic similarities between items, user behavior, and preferences. 
+This approach delivers personalized recommendations, improving user engagement and satisfaction.
+
+* Enhance the contextual relevance of your applications with Retrieval Augmented Generation (RAG) - RAG is a technique for enhancing the accuracy and reliability of generative AI models with facts fetched from external sources that are contextual to your application, such as an internal vector database.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:34]: change in capitalization here is distracting as well

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 20]
```diff
@@ -0,0 +1,227 @@
...
+This method shows how you can create the default vector index configuration using the Couchbase Lite Vector Search extension.
+
+[source, csharp]
+----
+
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-create-default-config]
+
+----
+
+First, we initialize the `config` object with a new `VectorIndexConfiguration` object intialized with the following parameters:
```
Comment (dgnicholson) [inline modules/csharp/pages/working-with-vector-search.adoc]: Nit
```suggestion
First, we initialize the `config` object with a new `VectorIndexConfiguration` object initialized with the following parameters:
```

File: modules/csharp/pages/vector-search.adoc [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+This results in the LLM returning a result that is likely to be far more contextually relevant than the standalone prompt.
+
+Additionally, Vector Search in Couchbase Lite provides the following benefits:
+
+* Unified Cloud-to-Edge Support for Vector Similarity Search - Couchbase supports Vector Search from cloud to edge, which enables applications to efficiently utilize cloud and edge computing's strengths.
+
+* Enhanced Data Privacy on the Edge - By performing Vector Search within the device, personal data and search queries of a sensitive nature do not have to leave the device.
+
+* Low Latency Application Support - You can run searches locally against a local dataset using a local embedded model. This eliminates the network variability and results in more consistent execution speed.
+Even in the case where the model is not embedded within the local device but is deployed at the edge location, the round trip time (RTT) associated with queries can be significantly reduced compared to searches made over the Internet.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:45]: "Even in a case..." ? indefinite article here?

File: modules/csharp/pages/vector-search.adoc (hunk 17 above)
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:47]: cloud model vs. cloud based model? are they the same or two different things?

**Review by dgnicholson — APPROVED** [review]
A few non-blocker nits, but good to go. Mostly just checked to make sure links and includes worked, as well as very high-level spelling/style check.

File: modules/csharp/pages/vector-search.adoc [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+By running queries locally on the device, you can save on data transfer costs and cloud egress charges while also decentralizing the operational costs.
+
+== Key Concepts in Vector Search
+
+When xref:csharp:working-with-vector-search.adoc[working with Vector Search,] you should be aware of the core concepts below. 
+
+=== About Vector Embeddings
+
+Vector embeddings represent the output of a Machine Learning (ML) model as an array of numbers to capture semantic or contextual relationships between data points.
+This representation encodes how a ML model understands the input or inputs provided to it, based on how the model was initially trained and the internal structure of the model.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:57]: a ML model? an ML model? I dunno

File: modules/csharp/pages/vector-search.adoc [hunk 23]
```diff
@@ -0,0 +1,137 @@
...
+When a model considers the features of a given input as similar, the distance between the vector embeddings will be short.
+Vector embeddings are stored within embedded vector indexes.
+
+=== About Vector Indexes
+
+Vector Indexes are used to store and manage vector representations of content in the form of vector embeddings.
+You can use Vector Indexes to efficiently retrieve vectors, similar to a target vector.
+Before use, a Vector Index needs to be trained to compute the centroids and parameters for encoding the vectors. 
+Training is configured to commence automatically on the first Vector Search query when there are enough vectors to be trained based on the `minimum-training-size` configuration.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:66]: maybe, 

Training begins automatically for the first Vector Search query that meets the minimum necessary number of vector. 
The minimum is set by the `minimum-training-size` configuration parameter.

File: modules/csharp/pages/vector-search.adoc [hunk 24]
```diff
@@ -0,0 +1,137 @@
...
+=== About Vector Encoding
+
+Vector encoding reduces the size of Vectors index by algorithmic compression.
+You can configure the Vector Encoding in Couchbase Lite depending on your application's needs.
+
+This vector encoding compression reduces disk space required and I/O time during indexing and queries, but greater compression can result in inaccurate results in distance calculations.
+
+Vector Search for Couchbase Lite supports the following encoding algorithms:
+
+* None - This will return the highest quality results but at high performance and disk space costs
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:77]: "This returns..."
to match your other bullets

File: modules/csharp/pages/vector-search.adoc [hunk 25]
```diff
@@ -0,0 +1,137 @@
...
+
+* Scalar Quantizer - This reduces the number of bits used for each number in a vector. 
+The number of bits per component can be set to 4, 6, or 8 bits.
+The default setting in Couchbase Lite is 8 bits Scalar Quantizer or SQ-8.
+
+* Product Quantizer - This reduces the number of dimensions and bits per dimension. 
+It splits the vectors into multiple subspaces and performing scalar quantization on each space independently before compression.
+This can produce higher quality results than Scalar Quantization at the cost of greater complexity.
+
+NOTE: Quantizers are algorithmic processes that map input values from a larger set to output values in a smaller set, common quantization processes can include operations such as rounding and truncation.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:87]: break into two sentences at the comma ?

File: modules/csharp/pages/vector-search.adoc [hunk 26]
```diff
@@ -0,0 +1,137 @@
...
+Vector embeddings represent the output of a Machine Learning (ML) model as an array of numbers to capture semantic or contextual relationships between data points.
+This representation encodes how a ML model understands the input or inputs provided to it, based on how the model was initially trained and the internal structure of the model.
+When a model considers the features of a given input as similar, the distance between the vector embeddings will be short.
+Vector embeddings are stored within embedded vector indexes.
+
+=== About Vector Indexes
+
+Vector Indexes are used to store and manage vector representations of content in the form of vector embeddings.
+You can use Vector Indexes to efficiently retrieve vectors, similar to a target vector.
+Before use, a Vector Index needs to be trained to compute the centroids and parameters for encoding the vectors.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:65]: maybe link centroids to the section where they're defined below?

File: modules/csharp/pages/vector-search.adoc [hunk 27]
```diff
@@ -0,0 +1,137 @@
...
+* Product Quantizer - This reduces the number of dimensions and bits per dimension. 
+It splits the vectors into multiple subspaces and performing scalar quantization on each space independently before compression.
+This can produce higher quality results than Scalar Quantization at the cost of greater complexity.
+
+NOTE: Quantizers are algorithmic processes that map input values from a larger set to output values in a smaller set, common quantization processes can include operations such as rounding and truncation.
+
+=== About Centroids
+
+Centroids are vectors that function as the center point of a vector cluster within the data set.
+Each vector is then associated to the vector it is closest to by https://en.wikipedia.org/wiki/K-means_clustering[k-means clustering.]
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:92]: it seems like something happens to allow the vectors to pair up like this (the "is then" makes me wonder, what happened?)
Does identifying the centroid have this effect on the other vectors?

File: modules/csharp/pages/vector-search.adoc [hunk 28]
```diff
@@ -0,0 +1,137 @@
...
+Each Centroid is contained within a bucket along with its associated vectors, the greater amount of Centroids, the greater the potential accuracy of the model.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:93]: I'd break into two sentences at the first comma.
Each Centroid is contained within a bucket along with its associated vectors. 
The larger the number of Centroids, the greater the potential accuracy of the model.

File: modules/csharp/pages/vector-search.adoc [hunk 29]
```diff
@@ -0,0 +1,137 @@
...
+This will also incur a longer indexing time.
+
+Choosing Centroids in vector search involves trade-offs that can impact clustering effectiveness and search efficiency. The initial selection of Centroids, the number chosen, and their sensitivity to high dimensionality and outliers affect the quality of vector clustering.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:96]: aha! did you CHOOSE a centroid? or configure something for them to get chosen for you?

File: modules/csharp/pages/vector-search.adoc [hunk 30]
```diff
@@ -0,0 +1,137 @@
...
+
+The general guideline for the amount of Centroids is approximately the square root of the number of documents.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:98]: amount ==> number

File: modules/csharp/pages/vector-search.adoc [hunk 31]
```diff
@@ -0,0 +1,137 @@
...
+
+=== About Dimensions
+
+Vector dimensions describes the amount of numbers in a given vector embedding, commonly known as its' width.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:102]: does dimensions have to be plural here? "dimensions describes" doesn't scan well
amount ==> number 
no apostrophe after its.

File: modules/csharp/pages/vector-search.adoc [hunk 32]
```diff
@@ -0,0 +1,137 @@
...
+The greater the amount of dimensions, the greater accuracy of results, this also results in greater compute and memory costs and a increase in the latency of the search.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:103]: break into two sentences at the second comma.

File: modules/csharp/pages/vector-search.adoc [hunk 33]
```diff
@@ -0,0 +1,137 @@
...
+Vector dimensions are dependent on the LLM used to generate the Vector Embeddings. 
+
+NOTE: Couchbase Lite supports dimension sizes in the range of `2 - 2048`.
```
Comment (lamagnifica) [inline modules/csharp/pages/vector-search.adoc:106]: I would write out ...range from 2 to 2048.

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 34]
```diff
@@ -0,0 +1,227 @@
...
+This method shows how you can create the default vector index configuration using the Couchbase Lite Vector Search extension.
+
+[source, csharp]
+----
+
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-create-default-config]
+
+----
+
+First, we initialize the `config` object with a new `VectorIndexConfiguration` object initialized with the following parameters:
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:31]: we --> you

(use 2nd person not 1st)

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 35]
```diff
@@ -0,0 +1,227 @@
...
+
+* The expression of the data as a vector.
+
+* The width or `dimensions` of the vector index is set to `3`.
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:35]: no formatting on dimensions (unless they have to type that in somewhere?)

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 36]
```diff
@@ -0,0 +1,227 @@
...
+
+* The amount of `centroids` is set to `2`.
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:37]: same

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 37]
```diff
@@ -0,0 +1,227 @@
...
+This means that there will be two buckets with a single centroid each that gathers together similar vectors.
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:38]: This results in two buckets, each with a single centroid that acts to gather similar vectors together within that bucket.

? not sure I'm getting centroids...!

File: modules/csharp/pages/working-with-vector-search.adoc (hunk 35 above)
Comment (lamagnifica) [reply modules/csharp/pages/working-with-vector-search.adoc:35]: throughout

File: modules/csharp/pages/working-with-vector-search.adoc (hunk 34 above)
Comment (lamagnifica) [reply modules/csharp/pages/working-with-vector-search.adoc:31]: throughout

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 38]
```diff
@@ -0,0 +1,227 @@
...
+
+There are also some default values that we did not set with the `VectorIndexConfiguration()` method, these are:
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:40]: change to two sentences at the comma

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 39]
```diff
@@ -0,0 +1,227 @@
...
+The method below shows the generation of a custom vector index configuration.
+
+[source, csharp]
+----
+
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-create-custom-config]
+
+----
+
+NOTE: The number of vectors, the width or dimensions of the vectors and the training size can incur high CPU and memory costs as the size of each variable increases.
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:57]: comma after ...dimensions of the vectors,...

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 40]
```diff
@@ -0,0 +1,227 @@
...
+
+|===
+
+CAUTION: Altering the default training sizes could be detrimental to the accuracy of returned results produced by the model and total computation time.
+
+== Generating Vectors
+
+You can use two methods to generate vectors in Couchbase Lite:
+
+. You can call a Machine Learning(ML) model, and embed the generated vectors inside the documents.
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:119]: space before (ML)

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 41]
```diff
@@ -0,0 +1,227 @@
...
+
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-create-index]
+
+----
+
+. First, create the standard configuration, setting up an expressions, number of dimensions and number of centroids for the vector embedding.
+
+. Next, create a collection variable passing it the contents of a given collection from our `database` object, in our case, the `colors` collection.
+
+. Finally, create a vector index, `colors_index`, on a collection and pass it our configuration.
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:140]: ...pass it ^to^ your...

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 42]
```diff
@@ -0,0 +1,227 @@
...
+
+[source, csharp]
+----
+        
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-create-predictive-index]
+
+----
+
+NOTE: You can use less storage by using the `prediction()` function as the encoded vectors will only be stored in the index. 
+However, the index time will be longer as vector embedding generation is occurring at run time.
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:155]: is occurring at --> occurs at

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 43]
```diff
@@ -0,0 +1,227 @@
...
+|Required
+|The target vector expression that returns a vector in the form of an array of numbers.
+|limit
+|Optional
+|The limit number of the returned matched results.
+The maximum number allowed is 10000. An error will be returned when creating a query with a limit greater than 10000.
+
+|===
+
+NOTE: The default value for the `limit` parameter is 3.
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:180]: why not put this up in the table with the rest of the info about limit?

File: modules/csharp/pages/working-with-vector-search.adoc [hunk 44]
```diff
@@ -0,0 +1,227 @@
...
+
+[source, csharp]
+----
+        
+include::csharp:example$code_snippets/VectorSearch.cs[tags=vs-use-vector-match]
+
+----
+
+This function performs vector search against a specific vector index identifier for the specified vector expression.
+If the specified index does not exist, an error will occur on creation of the query.
```
Comment (lamagnifica) [inline modules/csharp/pages/working-with-vector-search.adoc:192]: If the specified index does not exist, creating the query results in an error.

(I tried to think of how to reword this somewhere above, just kind of got it now. 
Try to minimize the passive/future even though it's wicked hard)

File: modules/csharp/partials/release-notes/couchbase-mobile-csharp-release-note.3.2.0.adoc [hunk 45]
```diff
@@ -1,29 +1,165 @@
 [#maint-3-2-0]
-== 3.2.0 Beta 1 -- March 2024
```
Comment (lamagnifica) [inline modules/csharp/partials/release-notes/couchbase-mobile-csharp-release-note.3.2.0.adoc:2]: you don't want to keep the beta 1 release notes?
Oh, I see, they were completely lacking in useful info!

**Review by lamagnifica — COMMENTED** [review]
finished the working with topic

**Comment by lamagnifica** [comment]
Wow, complicated stuff! Nice job collecting all the info, @ElliotFrancisHunter

File: modules/csharp/partials/release-notes/couchbase-mobile-csharp-release-note.3.2.0.adoc (hunk 45 above)
Comment (borrrden) [reply modules/csharp/partials/release-notes/couchbase-mobile-csharp-release-note.3.2.0.adoc:2]: There was no beta 1 release for .net
